tests/cross_fw/test_templates/test_weights_compression_backends.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+
+import pytest
+
+from nncf.experimental.common.tensor_statistics.collectors import HAWQAggregator
+from nncf.experimental.common.tensor_statistics.collectors import NoopReducer
+from nncf.experimental.common.tensor_statistics.collectors import TensorCollector
+
+
+class TemplateTestMixedPrecisionAlgoBackend:
+    @abstractmethod
+    def get_hawq_with_backend(self, subset_size: int):
+        """
+        Returns an instance of the algorithm with the specified subset size.
+
+        :param subset_size: The size of the subset to be used by the algorithm.
+        :return: An instance of the algorithm.
+        """
+
+    @pytest.mark.parametrize("subset_size", (10, 1, None))
+    def test_hawq_statistic_collector(self, subset_size: int):
```

Comment:
It's good to have tests for hawq stat collector. What about adding tests for others?

There are already tests for calling statistics and passing subset_size. https://github.com/openvinotoolkit/nncf/blob/develop/tests/openvino/native/quantization/test_weights_compression.py#L892-L918
you could just extend them with list of sensitivity criterion.

Reply:
extend to all criterion collectors